File: time.js
/*

** HOW TO USE **
NAME A VOICE CHANNEL "utctimebot"
USE THE "~~utc" COMMAND


*/

const discord = require("discord.js");
const bot = new discord.Client();
const fs = require("fs");
const token = JSON.parse(fs.readFileSync("./bottoken.json"));

var channelJson = fs.readFileSync("./channels.json");
var jsonData = JSON.parse (channelJson);
var interval;
var settingTime;

bot.login(token.token);

bot.on("ready",() =>{
    console.log("UTC Time Bot online");
	settingTime = false;
    interval = setInterval(function(){checkTime(jsonData);},180000);
});

bot.on('error',(err) => {
    console.log("bot error: " + err.message);
});

bot.on("message",(message)=>{
    if(message.content == "~~utc") {
        message.channel.send("adding channels in all servers...");
        var count = addChannels();
        message.channel.send("found " + count + " new channel(s)");
        message.channel.send("resuming bot...");
    }
});

function addChannels(){
    var count = 0;
    bot.guilds.cache.forEach((thisguild)=>{
        thisguild.channels.cache.forEach((thisch)=>{
            if(thisch.name == "utctimebot"){
                if(!jsonData.includes(thisch.id)){
                    jsonData.push(thisch.id);
                    count++;
                }
            }
        });
    });
    fs.writeFileSync("./channels.json",JSON.stringify(jsonData),null,4);
    clearInterval(interval);
    interval = setInterval(function(){checkTime(jsonData);},180000);
    return count;
}

function checkTime(jsonData){
    var date = new Date();
    var hours = date.getUTCHours();
    var minutes = date.getUTCMinutes();
    if (hours < 10) hours = "0" + hours;
    if (minutes < 10) minutes = "0" + minutes;
    var time = hours + ":" + minutes + " (UTC)";
	console.log("time is: ", time);
	if(settingTime == false){
		settingTime = true;
		bot.guilds.cache.forEach((thisguild)=>{
        thisguild.channels.cache.forEach((thisch)=>{
            if(jsonData.includes(thisch.id)){
				console.log("setting time...");
				thisch.setName("🕑 " + time).then(ch=>{
                    console.log("promise fulfilled: "+time + " channel: " + ch);
					settingTime = false;
                }).catch(err=>{console.log(err)});
            }
        });
    });
	}
}

File: time.py
import discord
import asyncio
import json
import datetime

with open("./bottoken.json") as token_file:
    token  = json.load(token_file)
settingTime = False
with open('./channels.json') as json_file:
    json_channels = json.load(json_file)

async def set_time(client):
    while True:
        time = datetime.datetime.utcnow()
        for channel in client.get_all_channels():
            if channel.id in json_channels:
                hours = str(time.hour)
                minutes = str(time.minute)
                if (len(hours) == 1): hours = "0" + hours
                if (len(minutes) == 1): minutes = "0" + minutes
                time_str = hours + ":" + minutes + " (UTC)"
                print("Time is: " + time_str)
                await channel.edit(name="🕐" + time_str)
                print("Time set!")
        await asyncio.sleep(30)

class Client(discord.Client):
    async def on_ready(self):
        print("UTCTime bot is online")

    async def on_message(self,message):
        if message.content == "~~utc":
            await message.channel.send("adding channels in all servers...")
            count = await self.add_channels()
            await message.channel.send("found " + str(count) + " new channel(s)")
            await message.channel.send("resuming bot...")
        
    async def add_channels(self):
        count = 0
        for channel in client.get_all_channels():
            if channel.name == 'utctimebot':
                if channel.id not in json_channels:
                    json_channels.append(channel.id)
                    count += 1
        with open('./channels.json','w') as json_file:
            json.dump(json_channels,json_file)
        return count
            


client = Client()
client.loop.create_task(set_time(client))
client.run(token)